impeller/renderer/backend/vulkan/compute_pass_vk.cc

```diff
@@ -213,8 +213,53 @@ bool ComputePassVK::BindResource(size_t binding,
   return true;
 }
 
+// Note:
+// https://github.com/KhronosGroup/Vulkan-Docs/wiki/Synchronization-Examples
+// Seems to suggest that anything more finely grained than a global memory
+// barrier is likely to be ignored. Confirming this on mobile devices will
+// require some experimentation.
+
+// |ComputePass|
+void ComputePassVK::AddBufferMemoryBarrier() {
+  vk::MemoryBarrier barrier;
+  barrier.srcAccessMask = vk::AccessFlagBits::eShaderWrite;
```

Comment:
These to me seem underspecified. Right now, this says "All commands before the barrier must have buffer-writes in their compute shaders be visible to buffer-reads in also the compute shaders of commands after the barrier".

The same can apply to transfer read-write (`VK_ACCESS_TRANSFER_READ|WRITE_BIT` in `VK_PIPELINE_STAGE_TRANSFER_BIT`) in the transfer stage in addition to any compute-render dependencies.

You can just `or` them up. But I suggest a comment indicating each value.

Reply:
This would also match up with Metal semantics implemented above.

Reply:
Why would I need to specify transfer dependencies in a barrier API that is specifically for compute-compute dependencies?

Reply:
I think I clarified on the API docs that it is only for compute-compute dependencies, but I can add a caveat that it does not cover other cases.

Reply:
Fair. But the first compute pass still needs to wait for say transfer writes on the buffers it works on to be visible. So my expectation would be that compute-compute dependencies while necessary are not sufficient.

Reply:
We don't use transfer commands anywhere yet though. my expectation is that if we needed this we would add this blit/transfer -> compute barrier functionality to the blit pass and not to the compute pass.